File: c/LeetCode_7_ReverseInteger.c
#include <stdio.h>
#include <stdlib.h>


int reverse(int x){
    int is_negative = (x < 0) ? 1 : 0;

    x = abs(x);
    int rev = 0;
    while (x != 0) {
        rev *= 10;
        rev += x % 10;
        x /= 10;
    }

    if (rev <= -2147483648 || rev >= 2147483647) {
        return 0;
    }

    return (is_negative) ? (rev * -1) : rev;
}


int main() {
    reverse(123);
    return 0;
}

File: python/LeetCode_7_ReverseInteger.py
def reverse(x: int) -> int:
    
    flag = 0
    if (x < 0):
        flag = 1
        x = abs(x) 
    
    rev = 0
    while (x != 0):
        rev *= 10
        rev += x % 10
        x //= 10
   
    if (rev <= -2147483648 or rev >= 2147483647):
        return 0

    return (rev * -1) if flag else rev 


reverse(-123)